tests/Anoma/Compilation/positive/test007.juvix

```diff
@@ -0,0 +1,27 @@
+-- pattern matching and lambda-case
+module test007;
+
+import Stdlib.Prelude open;
+import Stdlib.Debug open;
+
+map' {A B} (f : A → B) : List A → List B :=
+  \ {
+    | nil := nil
+    | (h :: t) := f h :: map' f t
+  };
+
+terminating
+map'' {A B} {{Partial}} (f : A → B) (x : List A) : List B :=
+  if (null x) nil (f (phead x) :: map'' f (tail x));
```

Comment:
Compiling this to Nockma relies on Core-level optimizations. The function `phead` can internally call a `fail` function which takes a string as argument. If `Partial` is not specialised and this internal call to `fail` is not inlined to directly call the builtin, you'll be translating strings and get a crash.

Reply:
We'd like to keep support for Strings so we can intercept messages from builtin fail: https://github.com/anoma/juvix/issues/2665

We've re-enabled support for PrimString in the Anoma target and refactored the failing tests to use `failwith` (i.e builtin fail) directly instead of via the Partial trait.

Reply:
I don't completely understand the rationale behind this replacement -- if you have strings in the Nockma backend you can use `Partial`, if you don't you can't use `failwith` either.

Reply:
> but we can support builtin fail because it's compiled directly to `OpFail` in Tree

I think that's not an entirely correct assumption, because of eta-expansions in the pipeline. So lambda-abstractions with string argument type might be added on top of the builtin, and the string argument would be applied to a lambda-function not to the builtin directly. Otherwise my checks for strings in `checkBuiltins'` wouldn't fail with only `failwith` because `OpFail` is already handled specially there. It might work because of subsequent optimisation which inlines the eta-expanded functions.

Reply:
I see. I'll replace the PrimString check and modify the tests to avoid using Partial/failwith.

Reply:
We've now modified the tests to avoid using Partial/failwith and restored the Strings check in the anoma backend.